### PR TITLE
feat: add support for creating and updating file contents

### PR DESCRIPTION
**Pull request by vootelerotov**
See https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents.

I applied the well known engineering principle of "monkey-see-monkey-do". Hopefully my grasp of how to do things in this project is not off too much. Checkstyle yelling at me was definitely helpful.

Happy to do smaller follow-ups :)

### CODE REVIEW

File: src/main/java/com/spotify/github/v3/repos/requests/FileCreate.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import javax.annotation.Nullable;
+
+/**
+ * Request to create a file.
+ */
+@Value.Immutable
+@GithubStyle
+@JsonSerialize(as = ImmutableFileCreate.class)
+@JsonDeserialize(as = ImmutableFileCreate.class)
+public interface FileCreate {
```
Comment (vootelerotov) [inline src/main/java/com/spotify/github/v3/repos/requests/FileCreate.java:37]: Went with request object, not sure where do you draw the line for sensible amount of parameters. Was on the fence with 3 here, with 4 for `FileUpdate` it seemed more clear cut.

File: src/test/java/com/spotify/github/v3/clients/RepositoryClientTest.java [hunk 2]
```diff
@@ -79,8 +89,10 @@
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
 import org.powermock.core.classloader.annotations.PrepareForTest;
 import org.powermock.modules.junit4.PowerMockRunner;
+import uk.co.datumedge.hamcrest.json.SameJSONAs;
```
Comment (vootelerotov) [inline src/test/java/com/spotify/github/v3/clients/RepositoryClientTest.java:95]: Saw it on classpath, okay to use? Do not want to worry about difference in whitespace/order in mock and real-deal.

**Comment by vootelerotov** [comment]
Bumping this up, anything I can do from my side to get this merged in?

**Comment by vootelerotov** [comment]
@dziemba, could you please tell me if there is anything else I could do to get this PR rolling?

Not fond of pulling in one specific person, but starting to think that this PR might have fell through the cracks :)

**Comment by ebk45** [comment]
Hi @vootelerotov.

We apologise for taking so long to get eyes on this PR, we haven't been able to maintain this library to the standard we would have liked to but this will be changing in the new year. If you could please get this rebased and we'll be prioritising all open PRs ready for review.

Thanks!

Ellie

**Comment by vootelerotov** [comment]
Hey @ebk45

Re-based the PR, it went rather smoothly.

I also noticed that there is another open PR (opened roughly a year ago) [here](https://github.com/spotify/github-java-client/pull/119) that aims to do the same thing as this one. I did not see it before.

Let me know if there is anything else I can do.

**Comment by vootelerotov** [comment]
> we haven't been able to maintain this library to the standard we would have liked to

Do not be too demanding with yourself, it takes a lot of work to maintain an OSS library, especially supporting features that you are not using yourself. I have used this wrapper in multiple small projects, and I have found it useful as-is.

> but this will be changing in the new year.
But glad to hear this :)

**Review by dennisgranath — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**